orchestrator/careplancontributor/service.go

```diff
@@ -172,43 +213,89 @@ func (s Service) createCarePlan(patient fhir.Patient) (*fhir.CarePlan, error) {
 	return &result, nil
 }
 
-func (s Service) createTask(serviceRequest fhir.ServiceRequest, patient fhir.Patient) (*fhir.Task, error) {
+func (s Service) createTask(serviceRequest fhir.ServiceRequest) (*fhir.Task, error) {
 	// Marshalling of Task is incorrect when providing input
 	// See https://github.com/samply/golang-fhir-models/issues/19
 	// So just create a regular map.
 
 	// TODO: Should we make new cross references here for requester, owner, service request and patient?
-	*serviceRequest.Id = "#serviceRequest-1"
-	*patient.Id = "#patient-1"
+
 	task := map[string]any{
 		"resourceType": "Task",
 		"status":       "requested",
 		"intent":       "order",
 		"requester":    serviceRequest.Requester,
 		"owner":        serviceRequest.Performer,
 		"reasonCode":   serviceRequest.Code,
-		// "for": fhir.Reference{
-		// 	Type:      to.Ptr("Patient"),
-		// 	Reference: patient.Id,
-		// },
-		// "input": []map[string]any{
-		// 	{
-		// 		"valueReference": fhir.Reference{
-		// 			Type:      to.Ptr("ServiceRequest"),
-		// 			Reference: serviceRequest.Id,
-		// 		},
-		// 	},
-		// },
-		// "contained": []interface{}{
-		// 	serviceRequest, patient,
-		// },
 	}
 	taskJSON, _ := json.MarshalIndent(task, "", "  ")
 	println(string(taskJSON))
 	createdTask, err := coolfhir.Workflow{CarePlanService: s.CarePlanService}.Invoke(task)
 	return createdTask, err
 }
 
+// When an application accepts the Task, we enrich the Task with more detailed information like the actual Patient and the ServiceRequest
+func (s Service) handleAcceptedTask(task *fhir.Task, serviceRequest *fhir.ServiceRequest, patient *fhir.Patient) (*fhir.Task, error) {
+
+	taskJSON, err := s.enrichTask(task, serviceRequest, patient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to enrich task: %w", err)
+	}
+
+	var enrichedTask fhir.Task
+	if err := s.CarePlanService.Update("Task/"+*task.Id, json.RawMessage(*taskJSON), &enrichedTask); err != nil {
```

Comment:
if `enrichTask` returns the `Task` (as map), you can simply pass it here and let the FHIR client do the marshalling).